lib/srv/desktop/audit.go

```diff
@@ -154,6 +154,51 @@ func (s *WindowsService) onSharedDirectoryAnnounce(sid string, m tdp.SharedDirec
 	s.auditCache.SetName(sessionID(sid), directoryID(m.DirectoryID), directoryName(m.Name))
 }
 
+// onSharedDirectoryAcknowledge emits a DesktopSharedDirectoryStart on a successful receipt of a
+// successful tdp.SharedDirectoryAcknowledge.
+func (s *WindowsService) onSharedDirectoryAcknowledge(
+	ctx context.Context,
+	emitter events.Emitter,
+	id *tlsca.Identity,
+	sid string,
+	desktopAddr string,
+	m tdp.SharedDirectoryAcknowledge,
+) {
+	if m.ErrCode != tdp.ErrCodeNil {
+		return
+	}
+
+	name, ok := s.auditCache.GetName(sessionID(sid), directoryID(m.DirectoryID))
+	if !ok {
+		name = events.UnknownEvent
```

Comment:
This confused me a little bit - you're using the constant just because it happens to be "unknown". I'd be okay with a literal "unknown" here.

Further - should we use a difference code here? Maybe one ending in `W`. If this happens, it means the internal state is corrupted and something's not right.

Reply:
> This confused me a little bit - you're using the constant just because it happens to be "unknown". I'd be okay with a literal "unknown" here.

👍

> Further - should we use a difference code here? Maybe one ending in W. If this happens, it means the internal state is corrupted and something's not right.

That makes some sense. I'm unclear on what the semantics are for these codes -- is that documented somewhere?

Reply:
I don't think so. If I had to guess, codes ending in `I` are informational and codes ending in `W` are warnings (errors).

Reply:
I'll add an additional PR to the top of this stack that adds warning codes.